### PR TITLE
drivers: i2c: Add a dump routine

### PR DESCRIPTION
**Pull request by sjg20**
Add a function to dump out a set of I2C messages. It uses debug logging
so it only useful for debugging.

### CODE REVIEW

File: drivers/i2c/i2c_common.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c);
+
+void i2c_dump_msgs(const char *name, const struct i2c_msg *msgs,
+		   uint8_t num_msgs, uint16_t addr)
+{
+	LOG_DBG("I2C msg: %s, addr=%x", name, addr);
+	for (unsigned int i = 0; i < num_msgs; i++) {
+		const struct i2c_msg *msg = msgs + i;
```
Comment (pabigot) [inline drivers/i2c/i2c_common.c:22]: I personally prefer this idiom, but the more common one in Zephyr is `&msgs[i]`.  I'm not asking that this change, but would like comments from others as to whether they would.

File: drivers/i2c/i2c_common.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Logging of I2C messages
+ *
+ * Copyright Google LLC
```
Comment (pabigot) [inline drivers/i2c/i2c_common.c:4]: I believe we need a year here.

**Review by pabigot — CHANGES_REQUESTED** [review]
I don't think this is going to see a lot of use, but I don't think it hurts, so I'm ok with it after fixing the copyright declaration.

File: drivers/i2c/i2c_common.c (hunk 2 above)
Comment (sjg20) [reply drivers/i2c/i2c_common.c:4]: done

File: drivers/i2c/i2c_common.c (hunk 1 above)
Comment (sjg20) [reply drivers/i2c/i2c_common.c:22]: changed

**Comment by sjg20** [comment]
> I don't think this is going to see a lot of use, but I don't think it hurts, so I'm ok with it after fixing the copyright declaration.

I don't like the way it is has turned out as it is much more verbose and less readable than what I originally wrote. But it is better than nothing and I'l already using it in the emulator.

**Review by pabigot — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**